docs(recipes): make uniform structure part 1 (#3853)
kubeshop/tracetest: docs/docs/examples-tutorials/recipes/running-tracetest-without-a-trace-data-store-with-manual-instrumentation.mdx

## Run This Quckstart Example

The example below is provided as part of the Tracetest project. You can download and run the example by following these steps:

Clone the Tracetest project and go to the Node.js Quickstart with Manual Instrumentation:

```bash
git clone https://github.com/kubeshop/tracetest
cd tracetest/examples/quick-start-nodejs-manual-instrumentation
```

Follow these instructions to run the included demo app and TypeScript example:

1. Copy the `.env.template` file to `.env`.
2. Log into the [Tracetest app](https://app.tracetest.io/).
3. This example is configured to use the OpenTelemetry Tracing Backend. Ensure the environment you will be utilizing to run this example is also configured to use the OpenTelemetry Tracing Backend by clicking on Settings, Tracing Backend, OpenTelemetry, Save.
4. Fill out the [token](https://docs.tracetest.io/concepts/environment-tokens) and [API key](https://docs.tracetest.io/concepts/agent) details by editing your `.env` file. You can find these values in the Settings area for your environment.
5. Run `docker compose up -d`.
6. Run tests from the Tracetest Web UI by accessing the app with the URL `http://app:8080/books`.

> Follow along with the sections below for an in detail breakdown of what the example you just ran did and how it works.

Reviewer: ```suggestion
Follow along with the sections below for a detailed breakdown of what the example you just ran did and how it works.
```

Enabling the tracer is done by preloading the trace file. As seen in the `package.json`.

```json
"scripts": {
  "app-with-grpc-tracer": "node -r ./tracing.otel.grpc.js app.js",
  "availability-with-grpc-tracer": "node -r ./tracing.otel.grpc.js availability.js",
},
```

## Running the Node.js App

To start the full setup, run the following command:

```bash
docker compose up -d
```

## Run Tracetest Tests

Open the URL and [start creating tests in the Web UI](https://docs.tracetest.io/web-ui/creating-tests)! Make sure to use the `http://app:8080/books` URL in your test creation, because your Node.js app and Tracetest Agent are in the same network.

Here's a sample of a failed test run, which happens if you use this selector and assertion pair.

Selector:
```
span[tracetest.span.type="general" name="Books List"]
```

Assertion:
```
attr: books.list.count = 4
```

![assertion](https://res.cloudinary.com/djwdcmwdz/image/upload/v1715607148/docs/app.tracetest.io_organizations_ttorg_e66318ba6544b856_environments_ttenv_56b66bc6e1a1cbbd_test_e9CCQuLSg_run_3_selectedAssertion_0_selectedSpan_d0c03aa5d02b9975_uqhhwl.png)

> It fails because of the `books.length` is equal to `3`.

Reviewer: ```suggestion
It fails because `books.length` is equal to `3`.
```
